topic-operator/src/main/java/io/strimzi/operator/topic/LabelPredicate.java

```diff
@@ -136,6 +140,9 @@ public static LabelPredicate fromString(String string) throws IllegalArgumentExc
 
     @Override
     public boolean test(HasMetadata resource) {
+        if (labels == null || labels.size() == 0) {
+            return true;
+        }
```

Comment:
My 2 cents on this change ...

Initially, I didn't make this suggestion because we are changing the meaning of the test in the predicate now.
When this line creates the `LabelPredicate` in the `Session` class of the topic operator:

```
LabelPredicate resourcePredicate = config.get(Config.LABELS);
```

I would expect that the predicate is true when the provided labels are verified and they are in the resource (the `KafkaTopic`).
It was the current behavior before this change. Now, if the checked resource (a `KafkaTopic` in this case) doesn't have the labels we passed to the `LabelPredicate` constructor, the predicate test returns true.
From my point of view, the previous behavior was the correct one, so I agreed on having the check for empty labels outside of it. Sure that the others have a different opinion on that @scholzj @tombentley